src/amoc_user.erl

```diff
@@ -30,17 +30,21 @@ init(Parent, Scenario, Id, State) ->
                 infinity -> repeat(F);
                 N -> repeat(F, N)
             end,
+            amoc_event:notify({test_end, Scenario}),
             normal
         catch
             throw:stop ->
+                amoc_event:notify({test_end, Scenario}),
                 normal;
             %% {R, get_stack()} will result in a compact error message
             %% {E, R, get_stack()} will result in a full stack report
             E:Reason ->
+                amoc_event:notify({test_crashed, Scenario}),
                 {E, {abnormal_exit, Reason}, erlang:get_stacktrace()}
         after
             ets:delete(amoc_users, Id)
         end,
+    amoc_event:notify({test_begin, Scenario}),
```

Comment:
This is not correct approach as at code will only be reached after a user performed a scenario.